library/src/main/java/com/novoda/downloadmanager/lib/DownloadTask.java

```diff
@@ -449,6 +453,12 @@ private void checkPausedOrCanceled() throws StopRequestException {
         }
     }
 
+    private void checkClientRules() throws StopRequestException {
+        if (!downloadReadyChecker.clientAllowsToDownload(originalDownloadBatch)) {
```

Comment:
I've been thinking more about this and it make less sense, and in fact it's lying to the client.

We'll be triggering rule checks on an invalidated batch. The original batch no longer represents the same batch as the one we have in progress.  

:x: